rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildConfigurationEndpointImpl.java

```diff
@@ -0,0 +1,198 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rest.endpoints;
+
+import org.jboss.pnc.dto.Build;
+import org.jboss.pnc.dto.BuildConfiguration;
+import org.jboss.pnc.dto.BuildConfigurationRef;
+import org.jboss.pnc.dto.BuildConfigurationRevision;
+import org.jboss.pnc.dto.GroupConfiguration;
+import org.jboss.pnc.dto.requests.BuildConfigWithSCMRequest;
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.dto.response.Parameter;
+import org.jboss.pnc.dto.response.TaskResponse;
+import org.jboss.pnc.facade.providers.api.BuildConfigurationProvider;
+import org.jboss.pnc.facade.providers.api.BuildConfigurationSupportedGenericParametersProvider;
+import org.jboss.pnc.facade.providers.api.BuildProvider;
+import org.jboss.pnc.facade.providers.api.GroupConfigurationProvider;
+import org.jboss.pnc.rest.api.endpoints.BuildConfigurationEndpoint;
+import org.jboss.pnc.rest.api.parameters.BuildParameters;
+import org.jboss.pnc.rest.api.parameters.BuildsFilterParameters;
+import org.jboss.pnc.rest.api.parameters.PageParameters;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.util.Optional;
+import java.util.Set;
+
+@Stateless
+public class BuildConfigurationEndpointImpl extends AbstractEndpoint<BuildConfiguration, BuildConfigurationRef> implements BuildConfigurationEndpoint {
+
+    private BuildConfigurationProvider buildConfigurationProvider;
+    private BuildProvider buildProvider;
+    private BuildConfigurationSupportedGenericParametersProvider bcSupportedGenericParametersProvider;
+    private GroupConfigurationProvider groupConfigurationProvider;
+
+    public BuildConfigurationEndpointImpl() {
+        super(BuildConfiguration.class);
+    }
+
+    @Inject
+    public BuildConfigurationEndpointImpl(BuildConfigurationProvider buildConfigurationProvider,
+                                          BuildProvider buildProvider,
+                                          BuildConfigurationSupportedGenericParametersProvider bcSupportedGenericParametersProvider,
+                                          GroupConfigurationProvider groupConfigurationProvider)  {
+
+        super (buildConfigurationProvider, BuildConfiguration.class);
+
+        this.buildConfigurationProvider = buildConfigurationProvider;
+        this.buildProvider = buildProvider;
+        this.bcSupportedGenericParametersProvider = bcSupportedGenericParametersProvider;
+        this.groupConfigurationProvider = groupConfigurationProvider;
+    }
+
+    @Override
+    public Page<BuildConfiguration> getAll(PageParameters pageParams) {
+        return super.getAll(pageParams);
+    }
```

Comment:
I wonder, would it work if we skip this implementation entirely? And the following that just call their super version too?

Reply:
This is doable if we use the `public` access for all the methods in `AbstractEndpoint`. Do you think if that's ok? I don't know the original reason why it was set to `protected`.

Reply:
The reason is probably that some endpoints don't have getAll. Eg. UserEndpoint.